### PR TITLE
chore(lint): enable ANN001 in ruff with line-level noqa for legacy callsites

### PR DESCRIPTION
**Pull request by ktinubu**
## Summary

- Adds `ANN001` (missing-type-function-argument) to ruff's `select` list in `pyproject.toml`.
- Silences the 126 existing violations across 20 files (95 unique callsites) with line-level `# noqa: ANN001` rather than per-file ignores, so any *new* missing-arg annotation is caught immediately.
- Updates `CLAUDE.md` and `.pre-commit-config.yaml` rule-list comments to match.

This is a ratchet, not a cleanup — legacy callsites stay un-annotated for now; future PRs can replace `# noqa: ANN001` with real type annotations one site at a time.

## Note on `S001`

`S001` was also requested but is not a valid ruff rule code — flake8-bandit codes in ruff start at `S101` (the `S` category is already enabled). Only `ANN001` was added. If `S001` was meant to refer to a different rule, please re-open with the correct code.

## Verification

- `uvx ruff check .` — `All checks passed!`
- `uvx ruff format --check .` — `93 files already formatted`
- pre-commit (full hook suite, including pyright/interrogate/docformatter) — all hooks passed on commit.

## Files touched (per-line `# noqa: ANN001` counts)

| File | Lines |
| --- | ---: |
| `src/utils/callbacks.py` | 40 |
| `tests/pipeline/test_schemas/test_config.py` | 16 |
| `src/models/components/vae.py` | 14 |
| `scripts/get_dataset_stats.py` | 4 |
| `tests/test_configs.py` | 4 |
| `src/models/ksin_flow_matching_module.py` | 3 |
| `src/models/components/sinkhorn.py`, `scripts/subdir_funtime.py` | 1 each |
| (12 other files) | 1 each |

Refs #212

## Test plan

- [ ] CI ruff job passes on this branch.
- [ ] `pr-metadata-gate.yaml` passes (linked issue #212 has type=Task, label=code-health, milestone=code-health v1.0.0; traces to Phase #158 → Epic #114).
- [ ] Confirm by inspection that no `[tool.ruff.lint.per-file-ignores]` entry for `ANN001` was added.

### CODE REVIEW

**Comment by ktinubu** [comment]
## Verification Results

All 11 CI checks pass on the latest commit (`83953e1`). Behavioral checks below were run locally on the PR worktree.

- [x] **`ruff check .` is clean** *(Level 2 — invokes the tool whose contract this PR changes)*
  ```bash
  $ uvx ruff check .
  All checks passed!
  ```

- [x] **Ratchet works: a NEW un-annotated arg outside the noqa'd lines is flagged** *(Level 2 — exercises the rule on synthetic input; this is the actual promise of the PR)*
  ```bash
  $ printf 'def canary_func(x):\n    return x + 1\n' > /tmp/ann001_canary.py
  $ uvx ruff check --select ANN001 /tmp/ann001_canary.py
  ANN001 Missing type annotation for function argument `x`
   --> /tmp/ann001_canary.py:1:17
    |
  1 | def canary_func(x):
    |                 ^
  2 |     return x + 1
  Found 1 error.
  ```

- [x] **No stale/wrong-line `# noqa: ANN001` introduced** *(Level 2 — runs ruff with RUF100 to detect unused noqa directives; the PR adds 95 noqa lines, so this catches misplacement)*
  ```bash
  $ uvx ruff check --extend-select RUF100 --output-format concise . | grep "ANN001"
  (no output — none of the 10 RUF100 hits in the repo are ANN001; all are pre-existing for SIM115/PLC0415/S603/S607/B017)
  ```

- [x] **`ANN001` is in `select` and absent from `[tool.ruff.lint.per-file-ignores]`** *(Level 5 — parses pyproject.toml; appropriate because the promise is "no per-file ANN001 ignores were added", which is a configuration policy with no executable form)*
  ```bash
  $ python3 -c "
  import re, pathlib
  text = pathlib.Path('pyproject.toml').read_text()
  m = re.search(r'\[tool\.ruff\.lint\.per-file-ignores\](.*?)(?=\n\[|\Z)', text, re.S)
  ann001_in_pfi = [l for l in m.group(1).splitlines() if 'ANN001' in l]
  sel = re.search(r'^select\s*=\s*(\[[^\]]+\])', text, re.M).group(1)
  print('per-file-ignores ANN001:', ann001_in_pfi)
  print('select:', sel)
  "
  per-file-ignores ANN001: []
  select: ["E", "F", "I", "S", "T", "UP", "W", "ANN001"]
  ```

- [x] **Pyright (1.1.409, matching CI) passes on the 3 files where pyright re-runs after this PR** *(Level 2 — runs the tool CI runs; needed because pyright 1.1.409 became stricter on `reportPrivateImportUsage` for torch and exposed 22 latent issues when the noqa change re-triggered the hook on these files)*
  ```bash
  $ PYRIGHT_PYTHON_FORCE_VERSION=1.1.409 pyright \
      src/utils/math.py \
      src/models/components/embed_pool.py \
      src/models/components/sinkhorn.py
  0 errors, 0 warnings, 0 informations
  ```

- [x] **Full CI on the PR head passes** *(Level 1 — the actual end-to-end check; this is the only level that exercises both the GitHub Actions runner environment and the pinned tool versions in `requirements-torch.txt`)*
  ```bash
  $ gh pr checks 682 --repo tinaudio/synth-setter
  Analyze (actions)              pass
  Analyze (python)               pass
  CodeQL                         pass
  check-pr-metadata              pass
  code-coverage                  pass
  code-quality                   pass     ← was failing on first push; fixed by 0e674d5 + 83953e1
  run_tests_conda                pass
  run_tests_macos (3.10)         pass
  run_tests_ubuntu (3.10)        pass
  run_tests_ubuntu (3.11)        pass
  ```

### What surfaced during verification
- The first push (`25927b8`) failed `code-quality` because pyright **1.1.409 in CI** flags `torch.div`, `torch.zeros`, `torch.exp`, `torch.cdist`, etc. under `reportPrivateImportUsage`, while my **local pyright 1.1.408** does not. The noqa change re-triggered pre-commit's pyright hook on `src/utils/math.py`, `src/models/components/embed_pool.py`, `src/models/components/sinkhorn.py` and exposed 23 pre-existing latent errors.
- Fixed in `0e674d5` + `83953e1` by adding line-level `# pyright: ignore[reportPrivateImportUsage]` to the 22 unique sites (no per-file pyright excludes added — same per-line policy as the ANN001 noqa).

File: pyproject.toml [hunk 1]
```diff
@@ -41,8 +41,9 @@ extend-exclude = []
 
 [tool.ruff.lint]
 # E=pycodestyle errors, F=pyflakes, I=isort, S=bandit security,
-# T=T10 (print) + T20 (debugger), UP=pyupgrade, W=pycodestyle warnings
-select = ["E", "F", "I", "S", "T", "UP", "W"]
+# T=T10 (print) + T20 (debugger), UP=pyupgrade, W=pycodestyle warnings,
+# ANN001=missing-type-function-argument
+select = ["E", "F", "I", "S", "T", "UP", "W", "ANN001"]
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply pyproject.toml:46]: Left as-is — the line-level `# noqa: ANN001` is intentional. The PR description frames this as a **ratchet**: enabling ANN001 globally + per-line noqa at existing sites means any *new* missing-arg annotation (in `src/`, `tests/`, or `scripts/`) is flagged immediately.

Switching `tests/**` to a `per-file-ignores` entry would silence both existing *and* new violations there, which defeats the ratchet for the whole tests tree. Issue #212's "encourage but don't gate" guidance is already satisfied by the per-line approach — existing untyped test args are tolerated (no churn), but the rule is enabled so new ones get caught. Uniform policy across `src/`, `tests/`, and `scripts/` is deliberate.

File: src/utils/math.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import torch
 
 
-def divmod(a, b):
-    return torch.div(a, b, rounding_mode="floor"), torch.remainder(a, b)
+def divmod(a, b):  # noqa: ANN001
+    return torch.div(a, b, rounding_mode="floor"), torch.remainder(a, b)  # pyright: ignore[reportPrivateImportUsage]
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply src/utils/math.py]: Left as-is — these `# pyright: ignore[reportPrivateImportUsage]` lines are intentional and the rationale is documented in [the verification comment on this PR](https://github.com/tinaudio/synth-setter/pull/682#issuecomment-4321020233).

**TL;DR:** Pyright **1.1.409** (pinned in CI via `requirements-torch.txt`) flags `torch.div`, `torch.zeros`, `torch.exp`, `torch.cdist`, etc. under `reportPrivateImportUsage`; pyright 1.1.408 (and earlier) does not. The ANN001 noqa change re-triggered the pyright pre-commit hook on this file and surfaced these pre-existing latent diagnostics. The fix landed in 0e674d5 + 83953e1.

A pyright-config or stub-override fix is out of scope for a one-rule ruff chore PR — and the per-line ignore matches the same per-line policy used for `# noqa: ANN001` in this PR (uniform ratchet, no file-wide opt-outs). Tracking a proper torch-stub fix as follow-up work would be reasonable, but not in this PR.

File: src/models/components/embed_pool.py [hunk 3]
```diff
@@ -5,14 +5,14 @@
...
-def make_sin_pos_enc(max_len, d_enc):
-    pe = torch.zeros(max_len, d_enc)
-    position = torch.arange(0, max_len, dtype=torch.float32).unsqueeze(1)
-    div_term = torch.exp(
-        torch.arange(0, d_enc, 2, dtype=torch.float32) * (-math.log(10000.0) / d_enc)
+def make_sin_pos_enc(max_len, d_enc):  # noqa: ANN001
+    pe = torch.zeros(max_len, d_enc)  # pyright: ignore[reportPrivateImportUsage]
+    position = torch.arange(0, max_len, dtype=torch.float32).unsqueeze(1)  # pyright: ignore[reportPrivateImportUsage]
+    div_term = torch.exp(  # pyright: ignore[reportPrivateImportUsage]
+        torch.arange(0, d_enc, 2, dtype=torch.float32) * (-math.log(10000.0) / d_enc)  # pyright: ignore[reportPrivateImportUsage]
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply src/models/components/embed_pool.py]: Left as-is, same root cause as [the math.py thread](https://github.com/tinaudio/synth-setter/pull/682#discussion_r3142843161) — pyright **1.1.409** in CI flags `torch.zeros`, `torch.arange`, `torch.exp`, `torch.sin`, `torch.cos` under `reportPrivateImportUsage` while local pyright 1.1.408 does not. Full rationale in [the verification comment](https://github.com/tinaudio/synth-setter/pull/682#issuecomment-4321020233); fix landed in 0e674d5 + 83953e1.

The 6 ignores in this file all stem from one underlying pyright/torch-stubs disagreement. Per-line ignores match the per-line `# noqa: ANN001` policy in this PR rather than introducing a separate file-wide pyright override. Resolving the root cause via stubs/config is reasonable follow-up work but out of scope for a one-rule ruff chore PR.

File: src/models/components/sinkhorn.py [hunk 4]
```diff
@@ -15,8 +15,8 @@ def sinkhorn_C(
 
     # we initialize our scaling vectors to ones
     num_tokens, num_params = costs.shape
-    u = torch.zeros(1, num_params, device=K.device, dtype=K.dtype)
-    v = torch.zeros(num_tokens, 1, device=K.device, dtype=K.dtype)
+    u = torch.zeros(1, num_params, device=K.device, dtype=K.dtype)  # pyright: ignore[reportPrivateImportUsage]
+    v = torch.zeros(num_tokens, 1, device=K.device, dtype=K.dtype)  # pyright: ignore[reportPrivateImportUsage]
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply src/models/components/sinkhorn.py]: Left as-is, same root cause as [the math.py thread](https://github.com/tinaudio/synth-setter/pull/682#discussion_r3142843161) — pyright **1.1.409** in CI flags `torch.zeros`, `torch.logsumexp`, `torch.exp`, `torch.cdist`, `torch.empty`, `torch.einsum` under `reportPrivateImportUsage` while local pyright 1.1.408 does not. Full rationale in [the verification comment](https://github.com/tinaudio/synth-setter/pull/682#issuecomment-4321020233); fix landed in 0e674d5 + 83953e1.

The 14 ignores in this file are all the same pyright/torch-stubs false-positive on standard `torch` constructors/ops. Per-line ignores match the per-line `# noqa: ANN001` policy in this PR rather than introducing a separate file-wide pyright override. Resolving the root cause via stubs/config is reasonable follow-up work but out of scope for a one-rule ruff chore PR.

File: src/utils/math.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ktinubu) [reply src/utils/math.py]: Update — fixed in 7503f0a. Now that #691 pinned pyright to 1.1.408 (the version that does *not* flag torch members under `reportPrivateImportUsage`), I dropped all 22 line-level `# pyright: ignore[reportPrivateImportUsage]` suppressions and widened the pyright exclude to all of `src/models/` (replacing the per-file entries in `pyrightconfig.json` and `.pre-commit-config.yaml`).

File: src/models/components/embed_pool.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ktinubu) [reply src/models/components/embed_pool.py]: Update — fixed in 7503f0a. Now that #691 pinned pyright to 1.1.408 (the version that does *not* flag torch members under `reportPrivateImportUsage`), I dropped all 22 line-level `# pyright: ignore[reportPrivateImportUsage]` suppressions and widened the pyright exclude to all of `src/models/` (replacing the per-file entries in `pyrightconfig.json` and `.pre-commit-config.yaml`).

File: src/models/components/sinkhorn.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ktinubu) [reply src/models/components/sinkhorn.py]: Update — fixed in 7503f0a. Now that #691 pinned pyright to 1.1.408 (the version that does *not* flag torch members under `reportPrivateImportUsage`), I dropped all 22 line-level `# pyright: ignore[reportPrivateImportUsage]` suppressions and widened the pyright exclude to all of `src/models/` (replacing the per-file entries in `pyrightconfig.json` and `.pre-commit-config.yaml`).
